src/main/java/com/google/sps/servlets/ListImagesServlet.java

```diff
@@ -0,0 +1,66 @@
+package com.google.sps.servlets;
+
+import com.google.api.gax.paging.Page;
+import com.google.cloud.storage.Blob;
+import com.google.cloud.storage.Bucket;
+import com.google.cloud.storage.Storage;
+import com.google.cloud.storage.StorageOptions;
+import com.google.gson.Gson;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+
+import java.lang.reflect.Type;
+import com.google.gson.JsonSyntaxException;
+import com.google.gson.reflect.TypeToken;
+
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+/** Shows all of the images uploaded to Cloud Storage. */
+@WebServlet("/list-images")
+public class ListImagesServlet extends HttpServlet {
+
+  @Override
+  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    // List all of the uploaded files.
+    String projectId = "summer22-sps-51";
+    String bucketName = "summer22-sps-51.appspot.com";
+    Storage storage = StorageOptions.newBuilder().setProjectId(projectId).build().getService();
+    Bucket bucket = storage.get(bucketName);
+    Page<Blob> blobs = bucket.list();
+
+
+    // List to store all the img html tag of the stored images
+    List<String> images_html = new ArrayList<String>();
+
+
+    // Adding cloud storage images as <img> html element to the list 
+    for(Blob blob : blobs.iterateAll()){
+        String imgTag = String.format("<img src=\"%s\" />", blob.getMediaLink());
+        images_html.add(imgTag);
+    }
+
+    // converting the list to json using gson
+    String images_json = convertToJsonUsingGson(images_html);
+
+    // Output images as list of json <img> elements as HTML.
+    response.setContentType("application/json;");
+    response.getWriter().println(images_json);
+  }
+
+
+  // convert to json using gson
+  private String convertToJsonUsingGson(List<String> images_html) {
+
+    Type listType = new TypeToken<List<String>>() {}.getType();
+
+    Gson gson = new Gson();
+    String json = gson.toJson(images_html, listType);
+
+    return json;
+  }
+}
```

Comment:
Did you test if it returned the json in a correct format? We might have to use a java class for the list instead of ArrayList

Reply:
haven't test it in this for our webpage but, it should work I did the same thing for my webpage, I not sure if there is a way to turn an array list to json but I know thats what works for a list.

Reply:
Alright, for my webpage  an ArrayList didn't work too well but maybe I got something else wrong. Let's merge this version first, I'll solve the conflict with mine next.

Reply:
I agree with Santiago here, as ArrayList seems to give JSON problems. Looks good to me though.

Reply:
I think the confusion is that this is an ArrayList which it's not, it's simply a List and which uses ArrayList for implementation. I used that on my webpage and it work if it doesn't work for this one I will fix it.